Update PackageOverrides.txt

ViktorHofer opened the pull request:
Update PackageOverrides.txt which ships inside the Microsoft.NETCore.App.Ref targeting pack with the latest state for .NET 9.

This improves build perf as assembly version checks are avoided for the packages listed with a version lower or equal.

Filed https://github.com/dotnet/runtime/issues/104639 for adding some automation.

I went through the entire list of inbox libraries from https://github.com/dotnet/runtime/blob/5795e8c056c14741b80dfa304a974f5c0d72432a/src/libraries/NetCoreAppLibrary.props#L28 and checked if there's a package for it and if so, added it to the list or updated its entry. I also updated all the existing entries with the latest package version.

ViktorHofer committed:
Update PackageOverrides.txt

Update PackageOverrides.txt which ships inside the Microsoft.NETCore.App.Ref targeting pack with the latest state for .NET 9.

This improves build perf as assembly version checks are avoided for the packages listed with a version lower or equal.

ViktorHofer commented on src/installer/pkg/sfx/Microsoft.NETCore.App/PackageOverrides.txt:
```diff
@@ -132,31 +154,37 @@ System.Runtime.Numerics|4.3.0
+System.Security.Cryptography.OpenSsl|5.0.0
 System.Security.Cryptography.Primitives|4.3.0
-System.Security.Cryptography.X509Certificates|4.3.0
-System.Security.Cryptography.Xml|4.4.0
```
From what I can see, this line was wrong as System.SEcurity.Cryptography.Xml was never inbox.

eerhardt replied:
Note that it is "inbox" in `Microsoft.AspNetCore.App`

ericstj replied:
It's in ASP.NET Core but not .NETCore.  Surprising that no-one ever hit this, maybe because the package version was old?  Or folks listen to us and don't actually use it.  I confirmed your analysis in 2.0 - 9.0.  Thanks for catching it and removing it.

ViktorHofer replied:
@eerhardt I see that an entry for it is missing in https://github.com/dotnet/aspnetcore/blob/main/eng/PackageOverrides.txt.

ericstj replied:
It's there: https://github.com/dotnet/aspnetcore/blob/96f53b4cd0d39c378adf18d908d493ac3a7f522e/eng/PackageOverrides.txt#L38

Not sorted 😕

eerhardt replied:
Looks like ASP.NET's should be updated to no longer include System.IO.Pipelines, since that is now in NETCore.App.

ViktorHofer replied:
@eerhardt will you take care of updating the PackageOverrides list in aspnetcore or do you want me to file an issue?

eerhardt replied:
I've opened https://github.com/dotnet/aspnetcore/issues/56724 to track updating aspnetcore's file.

ViktorHofer commented:
cc @eerhardt

eerhardt commented on src/installer/pkg/sfx/Microsoft.NETCore.App/PackageOverrides.txt:
```diff
@@ -132,31 +154,37 @@ System.Runtime.Numerics|4.3.0
+System.Text.Encoding.CodePages|9.0.0
 System.Text.Encoding.Extensions|4.3.0
-System.Text.RegularExpressions|4.3.0
+System.Text.Encodings.Web|9.0.0
+System.Text.Json|9.0.0
```
These `9.0.0` ones are interesting. Is there a reason they weren't included to begin with?

ViktorHofer replied:
System.Text.Json is a new package but System.Text.Encodings.Web existed since 2015 and has been inbox since at least netcoreapp2.0. If I'm not mistaken it was you who created the initial list?

ericstj replied:
This list hasn't changed since you originally added it.  I think we were letting platform manifest do the work then.

With PackageOverrides we can short circuit that work.  It can also help if we eventually feed this data into NuGet for supplied by platform (short circuiting even earlier in restore).

eerhardt replied:
> If I'm not mistaken it was you who created the initial list?

I believe it was initially added in https://github.com/dotnet/core-setup/pull/7369/.

ViktorHofer replied:
That was just copied from your list: https://github.com/dotnet/core-setup/pull/7369/files#diff-d43a6b44d056f972850070a59b2db1c58d30dd708de5b8a27f492a4b97d71210R4

ericstj commented on src/installer/pkg/sfx/Microsoft.NETCore.App/PackageOverrides.txt:
```diff
@@ -1,91 +1,97 @@
+runtime.ubuntu.16.10-x64.runtime.native.System.Security.Cryptography.OpenSsl|4.3.3
 System.AppContext|4.3.0
-System.Buffers|4.4.0
+System.Buffers|4.5.1
```
Rather than listing exact serviced versions like this, what if we instead list a higher version that will never need to update?   We know that this package will never win over the framework copy (unless we ship a much newer version) so why bother churning on these serviced versions?

I care less about stuff we no longer build, but for things we're still building like System.Buffers it might be better to just list a threshold; `4.6.0`, or `5.0.0`, or even `9.0.0`.

ericstj reviewed (commented):
We should consider what's more maintainable for packages that are still built in maintenance-packages or older runtime but not current runtime.

ViktorHofer committed:
Update packages which will ship again

Update packages that aren't produced from runtime/main anymore but from other branches or from the maintenance-packages repository.

ViktorHofer reviewed (commented):
I updated the package versions for ones that will ship again from maintenance-packages or from a servicing branch. Any other feedback?

ViktorHofer commented:
@ericstj / @eerhardt any more feedback? If not, would you mind approving the PR?

eerhardt commented on src/installer/pkg/sfx/Microsoft.NETCore.App/PackageOverrides.txt:
```diff
@@ -132,33 +154,39 @@ System.Runtime.Numerics|4.3.0
 System.Xml.XmlSerializer|4.3.0
 System.Xml.XPath|4.3.0
-System.Xml.XPath.XDocument|4.3.0
+System.Xml.XPath.XDocument|5.0.0
```
We haven't shipped a new version of `System.Xml.XPath.XDocument` since 2016. Why does it get `5.0.0` and the other System.Xml packages don't?

ViktorHofer replied:
This package and a few others (i.e. DispatchProxy that you commented on below) will soon get a new version for servicing. The sources got moved to https://github.com/dotnet/maintenance-packages/tree/main/src.

eerhardt commented on src/installer/pkg/sfx/Microsoft.NETCore.App/PackageOverrides.txt:
```diff
@@ -94,36 +100,52 @@ System.IO.Compression|4.3.0
-System.Reflection.Emit|4.3.0
-System.Reflection.Emit.ILGeneration|4.3.0
-System.Reflection.Emit.Lightweight|4.3.0
+System.Reflection.DispatchProxy|6.0.0
```
I think this one can stay at `5.0.0`, we never shipped a `5.0.0` (just 5.0.0-preview packages for a while), and it makes it consistent with the other 4.x packages.

eerhardt reviewed (approved):
LGTM. Just some minor questions/comments.

ericstj commented:
LGTM
